tests/unit_tests/grid/test_xyz.py

```diff
@@ -0,0 +1,656 @@
+
+    crs = rqc.Crs(basic_regular_grid.model,
+                  x_offset = x_offset,
+                  y_offset = y_offset,
+                  z_offset = z_offset,
+                  z_inc_down = True,
+                  rotation = rotation,
+                  rotation_units = rotation_units,
+                  xy_units = 'm',
+                  z_units = 'm',
+                  axis_order = 'easting northing',
+                  epsg_code = '32630')
+    crs.create_xml()
+
+    global_xy_units = 'm'
+    global_z_units = 'm'
+    global_z_increasing_downward = False
+
+    a = np.array([[0.5 + np.sqrt(3) / 2, 0.5 - np.sqrt(3) / 2,
+                   -1.0], [0.5 + np.sqrt(3) / 2, 0.5 - np.sqrt(3) / 2, -1.0],
+                  [0.5 + np.sqrt(3) / 2, 0.5 - np.sqrt(3) / 2, -1.0]])
+    a_expected = np.array([[1.0, 1.0, 1.0], [1.0, 1.0, 1.0], [1.0, 1.0, 1.0]])
+
+    # Act
+    a_transformed = basic_regular_grid.global_to_local_crs(a,
+                                                           crs_uuid = crs.uuid,
+                                                           global_xy_units = global_xy_units,
+                                                           global_z_units = global_z_units,
+                                                           global_z_increasing_downward = global_z_increasing_downward)
+
+    # Assert
+    np.testing.assert_array_almost_equal(a_transformed, a_expected)
+
+
+def test_global_to_local_crs_full_transformation(basic_regular_grid):
+    # Arrange
+    x_offset = -1.0
+    y_offset = 3.0
+    z_offset = 2.0
+    rotation = 45.0
+    rotation_units = 'dega'
+
+    crs = rqc.Crs(basic_regular_grid.model,
+                  x_offset = x_offset,
+                  y_offset = y_offset,
+                  z_offset = z_offset,
+                  z_inc_down = True,
+                  rotation = rotation,
+                  rotation_units = rotation_units,
+                  xy_units = 'm',
+                  z_units = 'm',
+                  axis_order = 'easting northing',
+                  epsg_code = '32630')
+    crs.create_xml()
+
+    global_xy_units = 'ft'
+    global_z_units = 'ft'
+    global_z_increasing_downward = False
+
+    a = np.array([[3.678873, 12.162424, -3.28084], [17.598297, 12.162424, -13.12336],
+                  [31.517722, 12.162424, -22.965879]])
+    a_expected = np.array([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0], [7.0, 8.0, 9.0]])
+
+    # Act
+    a_transformed = basic_regular_grid.global_to_local_crs(a,
+                                                           crs_uuid = crs.uuid,
+                                                           global_xy_units = global_xy_units,
+                                                           global_z_units = global_z_units,
+                                                           global_z_increasing_downward = global_z_increasing_downward)
+
+    # Assert
+    np.testing.assert_array_almost_equal(a_transformed, a_expected)
```

Comment:
Could also follow with the reverse transformation here, asserting end result is same as starting array